src/library/scala/concurrent/Promise.scala

```diff
@@ -60,19 +60,20 @@ trait Promise[T] {
    *
    *  @return   This promise
    */
-  final def completeWith(other: Future[T]): this.type = tryCompleteWith(other)
-
-  /** Attempts to complete this promise with the specified future, once that future is completed.
-   *
-   *  @return   This promise
-   */
-  final def tryCompleteWith(other: Future[T]): this.type = {
+   def completeWith(other: Future[T]): this.type = {
     if (other ne this.future) { // this tryCompleteWith this doesn't make much sense
       other.onComplete(this tryComplete _)(Future.InternalCallbackExecutor)
     }
     this
   }
 
+  /** Attempts to complete this promise with the specified future, once that future is completed.
+   *
+   *  @return   This promise
+   */
+  @deprecated("Since this method is semantically equivalent to `completeWith`, use that instead.", "2.13")
+  final def tryCompleteWith(other: Future[T]): this.type = completeWith(other)
```

Comment:
Why did `completeWith` get chosen as the name to keep, and not `tryCompleteWith`?

Reply:
@NthPortal Great question! My rationale was that `completeWith` is by necessity not always immediate, hence completeWith becomes an attempt by its very nature. I'm open to being convinced otherwise! :-)

Reply:
As it is right now, there are two pairs of methods: `(tryC|c)omplete` and `(tryC|c)ompleteWith`; my first intuition is that the methods of each pair should behave "the same", so to speak.

For the first pair, `tryComplete` does nothing if it fails, while `complete` throws an exception. I would expect then that `tryCompleteWith` does nothing when it fails (which is correct), and that `completeWith` throws an exception (which is *not* correct). Obviously, there is no way to have `completeWith` throw an exception, because it fails on a different thread, but its behaviour is still less immediately intuitive to me. I feel that `tryCompleteWith` better indicates what it does, and mirrors `tryComplete` more.

I would love to get broader feedback on what others feel is the more intuitive name to them.

Reply:
@NthPortal Sounds good. Let's see if there's more input from others here.